python/ml4ir/base/pipeline.py

```diff
@@ -390,6 +390,26 @@ def run(self):
                 experiment_tracking_dict, orient="index", columns=["value"]
             ).to_csv()
 
+            # temperature scaling
+            if self.args.execution_mode in {
+                ExecutionModeKey.TRAIN_INFERENCE_EVALUATE,
+                ExecutionModeKey.TRAIN_EVALUATE,
+                ExecutionModeKey.TRAIN_INFERENCE,
+                ExecutionModeKey.TRAIN_ONLY,
+                ExecutionModeKey.INFERENCE_EVALUATE_RESAVE,
+                ExecutionModeKey.EVALUATE_RESAVE,
+                ExecutionModeKey.INFERENCE_RESAVE,
+            }:
+                if 'calibration' in self.model_config:
+                    if self.model_config['calibration']['key'] == 'temperature_scaling':
+                        temperature = self.model_config['calibration']['temperature'] if \
+                            'temperature' in self.model_config['calibration'] else 1.5
+                        temperature_scale(relevance_model=relevance_model,
+                                          dataset=relevance_dataset,
+                                          logger=self.logger,
+                                          logs_dir_local=self.logs_dir_local,
+                                          temperature_init=temperature)
+
```

Comment:
@marjanhs I recommend moving this call inside of `RelevanceModel.calibrate()`. That will allow you to keep the logic of mapping model config to the calibration function contained to RelevanceModel. Keep in mind that customers might use RelevanceModel as is by importing without the pipeline. The pipeline is just an example of one such use.

Reply:
ok, changed accordingly.